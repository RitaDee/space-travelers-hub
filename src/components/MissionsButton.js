import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import styles from 'src/css/MissionsButton.module.css';

function MissionsButton({ isReserved }) {
  return (
    <button
      type="button"
      className={cn(
        styles.button,
        isReserved ? styles.leaveMission : styles.joinMission,
      )}
    >
      {isReserved ? 'Leave Mission' : 'Join Mission'}
    </button>
  );
}

MissionsButton.propTypes = {
  isReserved: PropTypes.bool.isRequired,
};

export default MissionsButton;
