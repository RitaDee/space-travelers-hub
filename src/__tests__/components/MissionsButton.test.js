import React from 'react';
import renderer from 'react-test-renderer';
import MissionsButton from '../../components/MissionsButton';

describe('MissionsButton', () => {
  const onClickMock = jest.fn();

  it('renders correctly when not reserved', () => {
    const component = renderer.create(
      <MissionsButton isReserved={false} onClick={onClickMock} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when reserved', () => {
    const component = renderer.create(
      <MissionsButton isReserved onClick={onClickMock} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls the onClick function when clicked', () => {
    const component = renderer.create(
      <MissionsButton isReserved={false} onClick={onClickMock} />,
    );
    const button = component.root.findByType('button');
    button.props.onClick();
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
