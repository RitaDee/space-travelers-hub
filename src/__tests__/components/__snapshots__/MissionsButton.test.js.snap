// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionsButton renders correctly when not reserved 1`] = `
<button
  className="button joinMission"
  onClick={[MockFunction]}
  type="button"
>
  Join Mission
</button>
`;

exports[`MissionsButton renders correctly when reserved 1`] = `
<button
  className="button leaveMission"
  onClick={[MockFunction]}
  type="button"
>
  Leave Mission
</button>
`;
